### ISSUE TITLE
CompressedImageCodec Dimension Options

### ISSUE THREAD

**Issue by seranotannason** [uber/petastorm#397]
For RGB images, CompressedImageCodec only supports the dimension (H, W, 3) whereas torchvision datasets seem to default to (3, H, W). Why isn't (3, H, W) supported as a dimension too?

**Comment by selitvin** [comment]
No particular reason. Is the goal to avoid extra dimensions reordering during the read? I think we can add a  parameter to the codec to be able to do that. Would be happy to accept a PR if you do that ;)

**Comment by seranotannason** [comment]
Yes, that's the goal, to make it more compatible with PyTorch. Most, if not all, PyTorch pretrained models accept images of (3, H, W) and it's just cumbersome to have to reshape it, especially since one may have to decode the image before being able to do so.

I'll be happy to do a PR, what parameter are you thinking of for the codec? I haven't seen why we need one.

**Comment by selitvin** [comment]
Oh, may be I misunderstood something. What are you proposing to do then? How do you propose to change petastorm behavior? Do you want to always return images as (3, H, W)? If that's the case, I am not sure it's a good idea since there are a lot of non pytorch users.

**Comment by seranotannason** [comment]
I'm looking at the following code:

```
if len(value.shape) == 2:
    # Greyscale image
    image_bgr_or_gray = value
elif len(value.shape) == 3 and value.shape[2] == 3:
    # Convert RGB to BGR
    image_bgr_or_gray = value[:, :, (2, 1, 0)]
else:
    raise ValueError('Unexpected image dimensions. Supported dimensions are (H, W) or (H, W, 3). '
                        'Got {}'.format(value.shape))
```

and I thought we would just add another condition to accept the (3, H, W) format

**Comment by selitvin** [comment]
If we accept (3, H, W) we would need to transpose it before sending data to `imencode` (otherwise it would fail). Then during decoding we would still get (H, W, 3), since this is how decode works... Am I missing something?

**Comment by seranotannason** [comment]
I see, I didn't know `imencode` has a constraint of (H, W, 3). Let me think about it again.
